core/templates/dev/head/editor/exploration_editor.html

```diff
@@ -154,16 +154,33 @@
         <button id="tutorialSaveButton" class="btn btn-default protractor-test-save-changes" ng-class="{'btn-success': isExplorationSaveable()}" ng-click="saveChanges()" ng-disabled="!isExplorationSaveable()">
           <span ng-if="!isSaveInProgress">
             <span ng-if="isPrivate()">
-              <i class="material-icons md-18 md-dark oppia-save-publish-button-icon"
-             alt="Save Draft">&#xE161;</i>
-              <span class="oppia-save-publish-button-label">Save Draft</span>
+              <span ng-if="!modalIsOpening">
+                <i class="material-icons md-18 md-dark oppia-save-publish-button-icon"
+               alt="Save Draft">&#xE161;</i>
+                <span class="oppia-save-publish-button-label">Save Draft</span>
+                <span class="hidden-sm hidden-xs" ng-if="getChangeListLength()" style="opacity: 0.5">(<[getChangeListLength()]>)</span>
+              </span>
+
+              <span ng-if="modalIsOpening">
+                <i class="oppia-save-publish-loading oppia-save-publish-button-icon"
+               alt="Loading">
+                 <span class="oppia-loading-dot-one">.</span>
```

Comment:
Actually, since this commit is quite small, I'm going to ask: can you make these three lines into a reusable directive, and use that instead? You could call the directive 'loadingDots' and put it in directives.js, and then it would be great if you could replace all instances of these three lines in the frontend with `<loading-dots></loading-dots>`. I think that would reduce duplication quite a bit!